httpd/Dockerfile

```diff
@@ -92,17 +86,23 @@ RUN set -x && \
     ${OPENSSL_PATH}/bin/openssl req -new -newkey ${SIG_ALG} -keyout pki/server.key -out pki/server.csr -nodes -subj "/CN=oqs-httpd" -config ${OPENSSL_CNF} && \
     ${OPENSSL_PATH}/bin/openssl x509 -req -in pki/server.csr -out pki/server.crt -CA cacert/CA.crt -CAkey cacert/CA.key -CAcreateserial -days 365
 
-## second stage: Only create minimal image without build tooling and intermediate build results generated above:
-FROM debian:buster-slim 
+# second stage: Only create minimal image without build tooling and intermediate build results generated above:
+FROM alpine
 # Take in global args
 ARG HTTPD_PATH
 ARG OPENSSL_PATH
-
-RUN apt-get update -qq \
-    && apt-get install -y libapr1-dev libaprutil1-dev
-
+#
+RUN apk add pcre-dev expat-dev
+#
 # Only retain the ${*_PATH} contents in the final image
 COPY --from=intermediate ${HTTPD_PATH} ${HTTPD_PATH}
+# copy over manually build libapr{util}
+COPY --from=intermediate /usr/local/apr/lib  /usr/local/apr/lib
+
+# Some size optimization:
+RUN rm -rf ${HTTPD_PATH}/bin/ab
```

Comment:
I'm not a Docker expert, but won't the effect of this be that the filesystem layer prior to the `RUN rm` command will still have ab in it?  Instead shouldn't you delete ab back in the intermediate image so it's never copied over?

Reply:
Yes, that's even better. Thanks!